src/components/select/multi-select/multi-select.test.tsx

```diff
@@ -46,60 +67,1111 @@ test("initially renders combobox with custom placeholder when placeholder prop i
+    const [value, setValue] = React.useState(["amber"]);
+
+    return (
+      <MultiSelect
+        label="Colour"
+        value={value}
+        onChange={(event) => {
+          setValue((event.target.value as unknown) as string[]);
+        }}
+      >
+        <Option text="amber" value="amber" />
+      </MultiSelect>
+    );
+  };
+  render(<ControlledMultiSelect />);
+
+  await user.click(screen.getByRole("combobox"));
+
+  const amberPill = await screen.findByTitle("amber");
+  const dimissButton = within(amberPill).getByRole("button");
+  await user.click(dimissButton);
+
+  expect(amberPill).not.toBeInTheDocument();
+});
+
+test("calls onChange prop when a previously selected option is removed", async () => {
+  const user = userEvent.setup();
+  const onChange = jest.fn();
+  const ControlledMultiSelect = () => {
+    const [value, setValue] = React.useState(["amber", "black"]);
+
+    return (
+      <MultiSelect
+        label="Colour"
+        value={value}
+        onChange={(ev) => {
+          setValue((ev.target.value as unknown) as string[]);
+          onChange(ev);
+        }}
+      >
+        <Option text="amber" value="amber" />
+        <Option text="black" value="black" />
+      </MultiSelect>
+    );
+  };
+  render(<ControlledMultiSelect />);
+
+  await user.click(screen.getByRole("combobox"));
+  await user.keyboard("{Backspace}");
+
+  expect(onChange).toHaveBeenCalledWith(
+    expect.objectContaining({
+      target: expect.objectContaining({
+        value: ["amber"],
+      }),
+    })
+  );
+});
+
+test("does not call onChange prop, when Backspace key is pressed but nothing has been selected", async () => {
+  const user = userEvent.setup();
+  const onChange = jest.fn();
+  const ControlledMultiSelect = () => {
+    const [value, setValue] = React.useState<string[]>([]);
+
+    return (
+      <MultiSelect
+        label="Colour"
+        value={value}
+        onChange={(ev) => {
+          setValue((ev.target.value as unknown) as string[]);
+          onChange(ev);
+        }}
+      >
+        <Option text="amber" value="amber" />
+        <Option text="black" value="black" />
+      </MultiSelect>
+    );
+  };
+  render(<ControlledMultiSelect />);
+
+  await user.click(screen.getByRole("combobox"));
+  await user.keyboard("{Backspace}");
+
+  expect(onChange).not.toHaveBeenCalled();
+});
+
+test("clears the input after clicking away from it", async () => {
+  const user = userEvent.setup();
   render(
     <>
-      <h2 id="my-select-heading">My Select</h2>
+      <MultiSelect label="Colour" onChange={() => {}}>
+        <Option text="amber" value="amber" />
+      </MultiSelect>
+      <p>Outside content</p>
+    </>
+  );
+
+  await user.type(screen.getByRole("combobox"), "a");
+  await user.click(screen.getByText("Outside content"));
+
+  expect(screen.getByRole("combobox")).toHaveValue("");
+});
+
+test("clears the input after an option is selected", async () => {
```

Comment:
question: won't the input always have an empty value as nothing has been typed here?